frontend/packages/kubevirt-plugin/src/components/create-vm/forms/create-vm-form.tsx

```diff
@@ -172,7 +172,7 @@ export const CreateVMForm: React.FC<CreateVMFormProps> = ({
 
   let source: React.ReactNode;
   let cdRom = false;
-  if (customSource?.dataSource) {
+  if (customSource?.dataSource && !sourceStatus) {
```

Comment:
this is strange, what is sourceStatus and why sourceStatus needed to be false here ?
maybe it need a better name ?
maybe it's a "quick and dirty" solution, it's ok, but then we need some comment to explain why we did this hack

Reply:
basically, what's going on here is that customSource = bootState (react context, only simple wizard, changes ONLY if form changes by user) sourceStatus = available boot source (external source).

Because bootstate has a default value of PVC as default it will never get all the values from sourceStatus, and will ALWAYS enter this IF statement. I'm guessing this label bug is an old bug discovered only after the testing for DS started. 
This will also fix https://github.com/openshift/console/pull/10956 btw.

If you will look on line 197, there is an ELSE IF statement, that will never be used but should be used in case there is an available boot source as the situation with DS.

Hope that clarifies it a bit. You can try to change those setting on master and see the difference.
I tried testing it with machines that have boot source and those without, seems to be working just fine and don't break the flow. you can also experiment and send me any feedback if you have.

Reply:
no, you just made it more confusing :-)
how is failing to create a vm related to ** this ** bug ?

not creating a VM is a major bug, a wrong label is a small one, if we have a prblem of not creating a VM and you know about it, you need to make sure everyone knows about it, and open a but with higher sevirty.

we have a different bug for the not creating a VM and it's already worked on.
if this line is for the other bug, please remove it from the fix of this one.

Reply:
The fix works for both issues, it wasn't intended for the bug of [#10956](https://github.com/openshift/console/pull/10956) , but it solve it as this fix make sure all the data pass, for this bug, is the pvcName, but when all data pass a machine can be created as well.

Reply:
> it wasn't intended for the bug

no, it was not, if you remove the offending line from the create code, a label is still presnted, so again, how is it connected to this bug ?
this is a big problem because this bug is labeled "medum" priority and only mention the label.

> what's going on here is that customSource = bootState (react context, only simple wizard, changes ONLY if form changes by user) sourceStatus = available boot source (external source).

The problem with this approach is that the code becomes more complex and less clear, if the added complexity is a hack and you intend to fix it better in the future, it need some comment, why this hack was made, and that you intend to write it clearer in the future.

Reply:
1. There is an empty label, no name of pvc which was the bug, this fixes it.

2. I'm not adding a hack, the complexity is due to the design of 2 sources of truth. one is the bootState and the second is sourceStatus.

Reply:
1. not true, did you check that you actually need to change the create backend call to fix a label ?
    i don't think it's necessary to touch a code that does API calls that create a VM in order to fix a lable.
    please check again.
2. exactly, the true fix will be to understand how we ended up with 2 sources of truth and fix the way
    sourceRef was  implemented, adding `if .. else ..` to solve a deeper problem is a hack.
    
    this extra check in the if is dangerous because it looks like an "if source is OK, render the source, o/w fallback to generic/error render" it's hard to understand that you highjecd the generic render to render the source in case of sourceRef.
    it creats a code that looks like:
    ```
    if source_ok:
         render source by type ( registry, url, pve )
    else: // a reader can assume source is not ok 
         render fallback // and it also renders source is sourceRef 
    ```

Reply:
1. I don't change the call, if you will look at the code the difference is the data population.
2. This is 2 years code, probable patches on patches with new features over time. not related to sourceRef in any way.

Reply:
1. no, see comment https://github.com/openshift/console/pull/10950#pullrequestreview-865732422
2.a. no, the support for source ref is new, so this condition leading to rendering sourceRef as a fallback instead of a type of PVC source can't be more then a few months.
2.b. even it it's 2 years old (it's not) the real fix will be to fix the source of the problem witch is the way sourceRef support is implemented, so it will not create this unexpected behavior in this file.

Reply:
I don't think you looked at the code at all, you can install the git lens extension it will tell you all about it. I don't see any point continue this discussion any further on this current platform, should be moved offline.